### PR TITLE
[FEATURE] Enable All ParameterBuilder and DomainBuilder classes to accept batch_list generically

### PR DESCRIPTION
**Pull request by alexsherstinsky**
### Scope
The ability to handle `batch_list` (multi-batch) generically by all `ParameterBuilder` and `DomainBuilder` classes is essential for building self-initializing expectations.  Existing unit tests have been updated (and made more robust).

Please annotate your PR title to describe what the PR does, then give a brief bulleted description of your PR below. PR titles should begin with [BUGFIX], [FEATURE],  [DOCS], or [MAINTENANCE]. If a new feature introduces breaking changes for the Great Expectations API or configuration files, please also add [BREAKING]. You can read about the tags in our [contributor checklist](https://docs.greatexpectations.io/en/latest/contributing/contribution_checklist.html).

Changes proposed in this pull request:
- JIRA: GREAT-464/GREAT-621
-
-

After submitting your PR, CI checks will run and @ge-cla-bot will check for your CLA signature.

For a PR with nontrivial changes, we review with both design-centric and code-centric lenses.

In a design review, we aim to ensure that the PR is consistent with our relationship to the open source community, with our software architecture and abstractions, and with our users' needs and expectations. That review often starts well before a PR, for example in github issues or slack, so please link to relevant conversations in notes below to help reviewers understand and approve your PR more quickly (e.g. `closes #123`).

Previous Design Review notes:
-
-
-

### Definition of Done
Please delete options that are not relevant.

- [x] My code follows the Great Expectations [style guide](https://docs.greatexpectations.io/docs/contributing/style_guides/code_style)
- [x] I have performed a [self-review](https://docs.greatexpectations.io/docs/contributing/contributing_checklist) of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added [unit tests](https://docs.greatexpectations.io/docs/contributing/contributing_test#writing-unit-and-integration-tests) where applicable and made sure that new and existing tests are passing.
- [x] I have run any local integration tests and made sure that nothing is broken.

Thank you for submitting!

### CODE REVIEW

File: great_expectations/rule_based_profiler/domain_builder/column_domain_builder.py [hunk 1]
```diff
@@ -11,21 +11,21 @@
...
-        batch: Optional[Batch] = None,
+        batch_list: Optional[List[Batch]] = None,
         batch_request: Optional[Union[BatchRequest, RuntimeBatchRequest, dict]] = None,
+        data_context: Optional["DataContext"] = None,  # noqa: F821
         column_names: Optional[List[str]] = None,
     ):
         """
         Args:
-            data_context: DataContext
+            batch_list: explicitly specified Batch objects foruse in DomainBuilder
```
Comment (NathanFarmer) [inline great_expectations/rule_based_profiler/domain_builder/column_domain_builder.py]: "for use"

File: great_expectations/rule_based_profiler/domain_builder/domain_builder.py [hunk 2]
```diff
@@ -106,47 +97,33 @@ def get_validator(
...
 
-    def _get_batch_ids(
+    def get_batch_ids(
         self,
         variables: Optional[ParameterContainer] = None,
     ) -> Optional[List[str]]:
         return get_batch_ids_from_batch_list_or_batch_request(
             data_context=self.data_context,
-            batch_list=[self.batch],
+            batch_list=self.batch_list,
```
Comment (NathanFarmer) [inline great_expectations/rule_based_profiler/domain_builder/domain_builder.py:115]: ❤️

File: great_expectations/rule_based_profiler/domain_builder/table_domain_builder.py [hunk 3]
```diff
@@ -1,11 +1,30 @@
...
+    def __init__(
+        self,
+        data_context: "DataContext",  # noqa: F821
+        batch_list: Optional[List[Batch]] = None,
+        batch_request: Optional[Union[BatchRequest, RuntimeBatchRequest, dict]] = None,
+    ):
+        """
+        Args:
+            data_context: DataContext
+            batch_list: explicitly specified Batch objects foruse in DomainBuilder
```
Comment (NathanFarmer) [inline great_expectations/rule_based_profiler/domain_builder/table_domain_builder.py]: "for use"

File: great_expectations/rule_based_profiler/domain_builder/column_domain_builder.py (hunk 1 above)
Comment (alexsherstinsky) [reply great_expectations/rule_based_profiler/domain_builder/column_domain_builder.py]: @NathanFarmer thank you!  Fixing in next commit.

File: great_expectations/rule_based_profiler/domain_builder/table_domain_builder.py (hunk 3 above)
Comment (alexsherstinsky) [reply great_expectations/rule_based_profiler/domain_builder/table_domain_builder.py]: @NathanFarmer thank you! Fixing in next commit.

**Review by NathanFarmer — APPROVED** [review]
LGTM other than two typos!

**Pull request merged**